skipper-go/contracts/src/Multihop.sol

```diff
@@ -48,24 +52,27 @@ contract Multihop is Ownable {
         IERC20(fromToken).transfer(route[0].pairAddress, fromAmount);
 
         // Loop through the route and execute the trades
-        for (uint256 i = 0; i < route.length; i++) {
-            DexHop memory hop = route[i];
+        uint256 length = route.length;
```

Comment:
Caching array length to avoid repeated loading from the state in each iteration.